src/main/java/chess/domain/state/MoveState.java

```diff
@@ -54,4 +57,32 @@ public Map<Position, PieceType> collectBoard() {
                         entry -> entry.getValue().pieceType()
                 ));
     }
+
+    public double calculateScore(final Color color) {
+        double totalScore = board.values().stream()
+                .filter(piece -> piece.isSameColor(color))
+                .mapToDouble(piece -> PieceScore.findScore(piece.pieceType()))
+                .sum();
```

Comment:
PieceScore 로부터 점수를 가져오는걸 piece 가 직접 하도록 변경해보면 어떨까요?
Piece 를 이용해서 로직을 만들 때 훨씬 편리할 것 같아요

```suggestion
        double totalScore = board.values().stream()
                .filter(piece -> piece.isSameColor(color))
                .mapToDouble(piece -> piece.score())
                .sum();
```

Reply:
각 기물이 score를 리턴하면 되겠군요! 좋은 방법인것 같아요. 👍